spec/rspec/core/rake_task_spec.rb

```diff
@@ -66,6 +66,11 @@ def spec_command
         task.rspec_opts = "-Ifoo"
         expect(spec_command).to match(/#{task.rspec_path}.*-Ifoo/)
       end
+
+      it 'wont add --pattern if rspec_opts includes --pattern' do
+        task.rspec_opts = "--pattern some_specs"
+        expect(spec_command).to match(/#{task.rspec_path} --pattern some_specs$/)
+      end
```

Comment:
On my initial read of this spec, I expected that it would pass even w/o your change -- after all, it's just showing that `--pattern some_specs` is included in the command, not that the default one is excluded.  After running it locally, I understand it now -- you are showing that the custom pattern comes after the rspec path and have an end-of-line anchor with `$` so it does show the default pattern not being included.  But it's a very implicit way to show it.  How about this instead:

``` ruby
expect(spec_command).to include("--pattern some_specs").and exclude(RSpec::Core::RakeTask::DEFAULT_PATTERN)
```

IMO, that's much more explicit and easy to understand.